### DISCUSSION TITLE
Feedback for Switch (Ubuntu Noble 24.04) and Steam deck

### DISCUSSION THREAD

**Discussion by Icing-weaponry** [vanilla-wiiu/vanilla#14] [Hardware Results]
In your video you mentioned, you didn't have either of those devices for testing so I thought it could be good to test it for you and report my findings.

Steam deck:

- Pro: I managed to pair and connect, I can even control the inputs and the menu using the mouse, buttons and touchscreen
- Con: the video feed isn't smooth, runs about 10-15 fps with a lot of dropped frames, it also freezes and the only fix is to disconnect and reconnect.

Switch:

- Pro: I can pair successfully
- Con: no image is displayed, the inputs also don't work

General observation:

- The Wii U itself doesn't seem to detect the game pad, an error message pops up after pairing that it's checking to update the gamepad firmware. It tried 3 times and then invites you to pair once again. During my testing the error message didn't seem to affect any functionality.
- The installation of dependencies was easy enough for ubuntu but for arch (steam deck) it was a bit more complex a lot of repositories within pacman seem to have unsecured keys prompting you to delete them. There's also the whole "read only" issue with the OS which will make the dependencies having to be reinstalled every time there's an update. If a Steam deck version is developed later down the line, an appimage would be recommended for ease of use.

Hopefully, this helps and if there are details missing that would allow me to help you, do let me know.

I think this project is absolutely amazing and I can't wait to see future progress.

**Comment by Midnite59** [comment]
Can confirm my switch did the exact same thing on Ubuntu Jammy 22.04

**Comment by viviannq** [comment]
I'm having different results on my Steam Deck, granted, its running Ubuntu 24.04. I can pair and send inputs to the Wii U, I just have no video.

**Comment by EMREOYUN** [comment]
For no video on Ubuntu, it might be related to issue #129. It somehow requires wired connection for my Ubuntu setup.
